src/internal.c

```diff
@@ -4918,9 +4918,9 @@ void SSL_ResourceFree(WOLFSSL* ssl)
 
         /* check if tracking stats */
         if (ctx_heap->flag & WOLFMEM_TRACK_STATS) {
-            XFREE(ssl_hint->stats, ssl->ctx->heap, DYNAMIC_TYPE_SSL);
```

Comment:
In the function InitSSL ssl_hint->stats and ssl->heap are created by using the ctx->heap that was used with creating the WOLFSSL structure. While ssl->heap should be the same as ssl->ctx->heap, freeing them with a potentially different heap pointer could cause the memory buckets to migrate from one pool to another.

Reply:
The reason for this change was a crash due to static memory alloc failure in InitSSL (with a pool size that was too small). Then it tries to do a SSL_ResourceFree and the ssl->ctx->heap was not populated. Maybe the better fix is to 1. make sure the ssl->ctx->heap is set earlier in InitSSL or 2. Add a ssl->ctx and ssl->ctx->heap check here. Thoughts?